### PR TITLE
Use NATIVE_ADDRESS_MARKER for senRequestNative

### PR DESCRIPTION
**Pull request by kostekIV**

### CODE REVIEW

File: eth/test/MostL2.js [hunk 1]
```diff
@@ -180,9 +166,7 @@ describe("MostL2", function () {
     });
 
     it("Emits correct event", async () => {
-      const { most, bazeroAdrressBytes } = await loadFixture(
-        deployEightGuardianMostFixture,
-      );
+      const { most } = await loadFixture(deployEightGuardianMostFixture);
```
Comment (Marcin-Radecki) [inline eth/test/MostL2.js:169]: in `deployEightGuardianMostFixture`, we still call `await most.addPair(bazeroAdrressBytes, bazeroAdrressBytes, false);`, but we don't do it anymore in `eth/scripts/1_l2_setup_bridge_contracts.js`. So not sure which version is correct

File: eth/contracts/MostL2.sol [hunk 2]
```diff
@@ -166,7 +168,7 @@ contract MostL2 is AbstractMost {
         );
 
         // transfer native
-        if (_destTokenAddress == bAzeroAddress) {
+        if (_destTokenAddress == NATIVE_MARKER_ADDRESS) {
```
[the remark this answers is not in the document]
Comment (kostekIV) [reply eth/contracts/MostL2.sol:167]: Good catch, I forgot to rm this require.

File: eth/test/MostL2.js (hunk 1 above)
Comment (kostekIV) [reply eth/test/MostL2.js:167]: The 1_l2_* is correct. Fixed

Comment (Marcin-Radecki) [reply eth/test/MostL2.js:167]: Do we need `bazeroAdrressBytes` at all returned by `deployEightGuardianMostFixture`? I mean in those tests, we call `receiveRequest` always with `bazeroAdrressBytes` as `destTokenAddress` argument. I mean, those tests pass right, but do they test native transfers?

Comment (kostekIV) [reply eth/test/MostL2.js]: Good catch again. Sorry I was doing it too fast :P. Anyway, fixed the tests (see the usage of abs)

File: azero/contracts/most-l2/lib.rs [hunk 3]
```diff
@@ -339,14 +338,9 @@ pub mod most_l2 {
                 .transferred_value(amount_to_bridge)
                 .invoke()?;
 
-            let dest_token_address = self
-                .supported_pairs
```
Comment (Marcin-Radecki) [inline azero/contracts/most-l2/lib.rs:343]: Do we want to remove this logic from contracts on AZERO side? https://github.com/Cardinal-Cryptography/most/pull/222/files#diff-ed93ab43297ff1acce2388424f1b571f305e5686aba3463a5d27d1b0393040eaR128

**Review by Marcin-Radecki — APPROVED** [review]

File: azero/contracts/most-l2/lib.rs (hunk 3 above)
Comment (kostekIV) [reply azero/contracts/most-l2/lib.rs:343]: This is removed here

**Pull request merged**